src/app/page.tsx

```diff
@@ -118,18 +119,31 @@ const Home: React.FC<HomeProps> = () => {
     },
   });
 
+  // Whether or not to filter out ordinals from the UTXOs
+  const [shouldFilterOrdinals, setShouldFilterOrdinals] = useState(true);
+
+  const [filterOrdinalsModalOpen, setFilterOrdinalsModalOpen] = useState(false);
+
+  const handleShouldFilterOrdinals = (value: boolean) => {
+    setShouldFilterOrdinals(value);
+  };
+
   // Fetch all UTXOs
   const {
     data: availableUTXOs,
     error: availableUTXOsError,
     isError: hasAvailableUTXOsError,
     refetch: refetchAvailableUTXOs,
   } = useQuery({
-    queryKey: [UTXO_KEY, address],
+    queryKey: [UTXO_KEY, address, shouldFilterOrdinals],
     queryFn: async () => {
       if (btcWallet?.getUtxos && address) {
         // all confirmed UTXOs from the wallet
         const mempoolUTXOs = await btcWallet.getUtxos(address);
+
+        // return the UTXOs if we don't need to filter out the ordinals
+        if (!shouldFilterOrdinals) return mempoolUTXOs;
+
         // filter out the ordinals
         const filteredUTXOs = await filterOrdinals(
```

Comment:
@gbarkhatov decompose `filterOrdinals` method and move logic for dust and ordinals to  AppState `/src/app/state/index.tsx`